// @heroku-cli/plugin-apps/lib/commands/regions/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _command = require('@heroku-cli/command');

var _cliUx = require('cli-ux');

class LabsDisable extends _command.Command {

  async run() {
    const sortBy = require('lodash.sortby');

    let { body: regions } = await this.heroku.get('/regions');
    if (this.flags.private) {
      regions = regions.filter(region => region.private_capable);
    } else if (this.flags.common) {
      regions = regions.filter(region => !region.private_capable);
    }
    regions = sortBy(regions, ['private_capable', 'name']);

    if (this.flags.json) {
      _cliUx.cli.styledJSON(regions);
    } else {
      _cliUx.cli.table(regions, {
        columns: [{ key: 'name', label: 'ID', format: n => _cliUx.cli.color.green(n) }, { key: 'description', label: 'Location' }, { key: 'private_capable', label: 'Runtime', format: c => c ? 'Private Spaces' : 'Common Runtime' }]
      });
    }
  }
}
exports.default = LabsDisable;
LabsDisable.topic = 'regions';
LabsDisable.description = 'list available regions for deployment';
LabsDisable.flags = {
  json: _command.flags.boolean({ description: 'output in json format' }),
  private: _command.flags.boolean({ description: 'show regions for private spaces' }),
  common: _command.flags.boolean({ description: 'show regions for common runtime' })
};
